### PR TITLE
Variation grouping

### PR DESCRIPTION
**Pull request by alistat**
### Issue

On products that have color and size variations in Skroutz XML, a different product per color but not per size is requested.

WooCommerce generates for each variant product:
* Parent ID (post id) for the product
* A variation ID for each term's attribute combination.

Currently, the Skroutz Analytics plugin does not provide a way to generate a unique ID that complies with the Skroutz XML color specifications.

### Solution

Add a new Unique ID setting that combines the parent product ID and the specified grouping attribute terms: 
`<parent product ID>-<grouping attribute term1 ID>-...-<grouping attribute termN ID>`

One or more grouping attributes can be set in the plugin settings. The order in which the attributes are set in the settings also affects the generated Unique ID. 

If a product is not a variant the `<product ID>` is used.
If a product is a variant and does not have one of the declared color attributes the `<parent product ID>` is used.
If a product is a variant and has one of the specified grouping attributes the `<parent product ID>-<attribute term ID>` is used.
If a product is a variant and has two of the specified grouping attributes the `<parent product ID>-<attribute1 termA ID>-<attribute2 termB ID>` is used.

The existing settings regarding the Product ID/Product SKU and the custom ID affect the `<product ID>` and the `<parent product ID>` as well.

----

### Example

For example let's assume we have 2 attributes and 2 terms for each attribute: 
* attribute: **color** (ID: 51), term1: **red** (ID: 61), term2: **black** (ID: 62)
* attribute: **size** (ID: 52), term1: **small** (ID: 63), term2: **large** (ID: 64)

If we create a variable product and all the possible variations, **5** product IDs are generated:
* Parent product with ID: 1
* Variation product (black, small) with ID: 2
* Variation product (black, large) with ID: 3
* Variation product (red, small) with ID: 4
* Variation product (red, large) with ID: 5

In Skroutz Analytics plugin settings we enable the new Unique ID option and set only the `color` as the grouping attribute.

Now let's say a user buys our product in color red and size small.  Skroutz Analytics plugin will set the product_id of the line item to: `1-61`

----

### Admin settings

Screenshot 2019-11-20 at 11 55 56
Screenshot 2019-11-20 at 11 57 24

----

### Extra

* Bump Wordpress Tested up version to 5.3
* Bump WooCommerce Tested up version to 3.8
* Add filter `wc_skroutz_analytics_product_id_filter` for allowing the customization of the generated Unique ID.
* Add the plugin version as a data attribute in the script tag of the analytics tracking script, to easily extract the version.
* Camelize flavor in description of Shop Account ID.
* Remove JSON.stringify() as is no longer required (https://github.com/skroutz/analytics.js/pull/51)

### CODE REVIEW

File: includes/class-wc-skroutz-analytics-tracking.php [hunk 1]
```diff
@@ -48,10 +48,11 @@ public function __construct() {
 	public function output_analytics_tracking_script() {
 		$analytics_url = constant( "WC_Skroutz_Analytics_Flavors::".$this->settings->get_flavor()."_analytics_url" );
 		$analytics_object = constant( "WC_Skroutz_Analytics_Flavors::".$this->settings->get_flavor()."_analytics_object" );
+		$plugin_version = WC_Skroutz_Analytics::PLUGIN_VERSION;
 
 		$analytics_script = "
 		<!-- Skroutz Analytics WooCommerce plugin - v".WC_Skroutz_Analytics::PLUGIN_VERSION." -->
```
Comment (kosdiamantis) [inline includes/class-wc-skroutz-analytics-tracking.php]: Perhaps use `$plugin_version` and here.

File: includes/class-wc-skroutz-analytics-integration.php [hunk 2]
```diff
@@ -273,4 +296,160 @@ private function register_widgets() {
...
+						<?php echo wp_kses_post( $data['label'] ); ?>
+					</label>
+					<br/>
+					<?php echo $this->get_description_html( $data ); // WPCS: XSS ok. ?>
+				</fieldset>
+			</td>
+		</tr>
+		<?php
+
+		return ob_get_clean();
```
Comment (kosdiamantis) [inline includes/class-wc-skroutz-analytics-integration.php:377]: Could we use Heredoc or Nowdoc strings here instead of output buffering?

Comment (alistat) [reply includes/class-wc-skroutz-analytics-integration.php:377]: Within a Heredoc you cannot call a function or use control structures easily

File: includes/class-wc-skroutz-analytics-product.php [hunk 3]
```diff
@@ -111,4 +118,31 @@ private function get_parent_product() {
...
+	private function get_product_terms_suffix() {
+		$product_attributes = $this->product->get_attributes();
+
+		$terms = null;
+		foreach ($this->items_product_id_settings['grouping_attributes'] as $attribute_name) {
+			$taxonomy_name = wc_attribute_taxonomy_name($attribute_name);
+
+			if ( isset($product_attributes[$taxonomy_name]) ) {
+				$term = get_term_by( 'slug', $product_attributes[$taxonomy_name], $taxonomy_name );
+				if ( $term ) {
```
Comment (kosdiamantis) [inline includes/class-wc-skroutz-analytics-product.php]: [Minor] Perhaps one-line it?

File: includes/class-wc-skroutz-analytics-product.php [hunk 4]
```diff
@@ -111,4 +118,31 @@ private function get_parent_product() {
...
+					$terms .= "-{$term->term_id}";
+				}
```
Comment (MrWhizzy) [inline includes/class-wc-skroutz-analytics-product.php]: would it be better if we used the following?
`if ( $term ) $terms .= "-{$term->term_id}";`

**Review by kosdiamantis — APPROVED** [review]
+1 LGTM

**Pull request merged**